commands.js

```diff
@@ -2069,6 +2129,78 @@ var commands = exports.commands = {
+		if (!this.can('rawpacket')) return false;
+		// secret sysop command
+		room.add(target);
+	},
+
+	/*********************************************************
+	 * Help commands
+	 *********************************************************/
+
+	commands: 'help',
+	h: 'help',
+	'?': 'help',
+	help: function (target, room, user) {
+		target = target.toLowerCase();
+
+		// overall
+		if (target === 'help' || target === 'h' || target === '?' || target === 'commands') {
+			this.sendReply("/help OR /h OR /? - Gives you help.");
+		} else if (!target) {
+			this.sendReply("COMMANDS: /nick, /avatar, /rating, /whois, /msg, /reply, /ignore, /away, /back, /timestamps, /highlight");
+			this.sendReply("INFORMATIONAL COMMANDS: /data, /dexsearch, /movesearch, /groups, /faq, /rules, /intro, /tiers, /othermetas, /learn, /analysis, /calc (replace / with ! to broadcast. Broadcasting requires: + % @ & ~)");
+			if (user.group !== Config.groupsranking[0]) {
+				this.sendReply("DRIVER COMMANDS: /warn, /mute, /unmute, /alts, /forcerename, /modlog, /lock, /unlock, /announce, /redirect");
+				this.sendReply("MODERATOR COMMANDS: /ban, /unban, /ip");
+				this.sendReply("LEADER COMMANDS: /declare, /forcetie, /forcewin, /promote, /demote, /banip, /unbanall");
+			}
+			this.sendReply("For an overview of room commands, use /roomhelp");
+			this.sendReply("For details of a specific command, use something like: /help data");
+		} else {
+			var altCommandHelp;
+			var helpCmd;
+			var targets = target.split(' ');
+			if (typeof commands[target] === 'string') {
+				// If a function changes with command name, help for that command name will be searched first.
+				altCommandHelp = target + 'help';
+				if (altCommandHelp in commands) {
+					helpCmd = altCommandHelp;
+				} else {
+					helpCmd = commands[target] + 'help';
+				}
+			} else if (targets.length > 1 && typeof commands[targets[0]] === 'object') {
+				// Handle internal namespace commands
+				var helpCmd = targets[targets.length - 1] + 'help';
+				var namespace = commands[targets[0]];
+				for (var i = 1; i < targets.length - 1; i++) {
+					if (!namespace[targets[i]]) return this.sendReply("Help for the command '" + target + "' was not found. Try /help for general help");
+					namespace = namespace[targets[i]];
+				}
+				if (typeof namespace[helpCmd] === 'object') {
+					return this.sendReply(namespace[helpCmd].join('\n'));
+				} else if (typeof namespace[helpCmd] === 'function') {
+					return this.parse('/' + targets.slice(0, targets.length - 1).concat(helpCmd).join(' '));
+				} else {
+					return this.sendReply("Help for the command '" + target + "' was not found. Try /help for general help");
+				}
+			} else {
+				helpCmd = target + 'help';
+			}
+			if (helpCmd in commands) {
+				if (typeof commands[helpCmd] === 'function') {
+					// If the help command is a function, parse it instead
+					this.parse('/' + helpCmd);
+				} else if (typeof commands[helpCmd] === 'object') {
```

Comment:
`typeof` doesn't distinguish between `{}`-objects and anything else that isn't a primitive, including arrays. Specifically, this can lead to confusion with namespaces.

You're looking for `Array.isArray`